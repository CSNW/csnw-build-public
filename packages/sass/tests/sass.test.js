const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const sass = require('../')();
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await sass.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.{scss,sass}', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
