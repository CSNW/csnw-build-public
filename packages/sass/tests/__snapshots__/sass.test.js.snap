// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/sass/tests/fixtures",
    "contents": "body {
  background-color: blue; }

h1 {
  color: red; }

",
    "cwd": "C:/dev/csnw-build/packages/sass/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/sass/tests/fixtures/a.css",
    "sourceMap": Object {
      "file": "a.css",
      "mappings": "AAEA,AAAA,IAAI,CAAC;EACH,gBAAgB,EAHP,IAAI,GAId;;ACJD,AAAA,EAAE,CAAC;EACD,KAAK,EAAE,GAAG,GACX",
      "names": Array [],
      "sources": Array [
        "../../../../a.scss",
        "_b.scss",
      ],
      "sourcesContent": Array [
        "$color-bg: blue;

body {
  background-color: $color-bg;
}

@import 'b';
",
        "h1 {
  color: red;
}
",
      ],
      "version": 3,
    },
  },
  "c.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/sass/tests/fixtures",
    "contents": "h2 {
  color: green; }

",
    "cwd": "C:/dev/csnw-build/packages/sass/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/sass/tests/fixtures/c.css",
    "sourceMap": Object {
      "file": "c.css",
      "mappings": "AAAA,AAAA,EAAE,CAAC;EACD,KAAK,EAAE,KAAK,GAAG",
      "names": Array [],
      "sources": Array [
        "../../../../c.sass",
      ],
      "sourcesContent": Array [
        "h2 {
  color: green; }
",
      ],
      "version": 3,
    },
  },
}
`;
